rust/agama-dbus-server/src/network/model.rs

```diff
@@ -324,6 +345,55 @@ impl PartialEq for BaseConnection {
     }
 }
 
+#[derive(Debug, Error)]
+#[error("Invalid MAC address: {0}")]
+pub struct InvalidMacAddress(String);
+
+#[derive(Debug, Clone)]
+pub enum MacAddress {
+    MacAddress(macaddr::MacAddr6),
+    Preserve,
+    Permanent,
+    Random,
+    Stable,
+}
```

Comment:
This is what I mean. Since these values may be exclusive to NM this may be too NM specific.

Reply:
That's fine with me. If we decide to add support for another backend in the future, we might want to change that. But by now, let's be pragmatic.